receipt/verify.go

```diff
@@ -0,0 +1,100 @@
+// verify handles the receipt verification.
+package receipt
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const (
+	// API endpoint for sandbox environment.
+	SANDBOX_URL = "https://sandbox.itunes.apple.com/verifyReceipt"
+	// API endpoint for production environment.
+	PRODUCTION_URL = "https://buy.itunes.apple.com/verifyReceipt"
+	// Request content-type for apple store.
+	CONTENT_TYPE = "application/json; charset=utf-8"
+)
+
+// Returns new IAP request with defult client
+func New() *Client {
+	return &Client{
+		HttpClient: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+// Returns new IAP request with given client
+func NewWithClient(client httpClient) *Client {
+	return &Client{
+		HttpClient: client,
+	}
+}
+
+// Verify receipts and gets result from app store endpoints
+func (client *Client) Verify(ctx context.Context, req IAPRequest) (response *IAPResponse, err error) {
```

Comment:
Apart from this, is it feasible to provide individual verification for sandbox/production?

Reply:
No.
Reason: Receipt is base64 encoded data,  we can not predict the environment(sandbox/production) from it.
Apple doc also says it, 
`As a best practice, always call the production URL for verifyReceipt first, and proceed to verify with the sandbox URL if you receive a 21007 status code.`